src/execution_plan/ops/op_expand_into.c

```diff
@@ -68,26 +68,28 @@ static void _populate_filter_matrix(OpExpandInto *op) {
  * clears filter matrix. */
 static void _traverse(OpExpandInto *op) {
 	// Create both filter and result matrices.
-	if(op->F == GrB_NULL) {
-		size_t required_dim = Graph_RequiredMatrixDim(op->graph);
-		GrB_Matrix_new(&op->M, GrB_BOOL, op->recordsCap, required_dim);
-		GrB_Matrix_new(&op->F, GrB_BOOL, op->recordsCap, required_dim);
+	size_t required_dim = Graph_RequiredMatrixDim(op->graph);
+	if(op->f == GrB_NULL) {
+		assert(GrB_Vector_new(&op->f, GrB_UINT64, op->recordsCap) == GrB_SUCCESS);
+		assert(GrB_Matrix_new(&op->M, GrB_BOOL, op->recordCount, required_dim) == GrB_SUCCESS);
 	}
 
 	// Populate filter matrix.
-	_populate_filter_matrix(op);
+	_populate_filter_vector(op);
 	// Clone expression, as we're about to modify the structure with Optimize.
 	AlgebraicExpression *clone = AlgebraicExpression_Clone(op->ae);
 	// Append filter matrix to algebraic expression, as the left most operand.
-	AlgebraicExpression_MultiplyToTheLeft(&clone, op->F);
+	AlgebraicExpression_VectorMultiplyToTheLeft(&clone, op->f);
 	// TODO: consider performing optimization as part of evaluation.
 	AlgebraicExpression_Optimize(&clone);
+	// Make sure M dimensions matches number of source nodes.
+	assert(GxB_Matrix_resize(op->M, op->recordCount, required_dim) == GrB_SUCCESS);
```

Comment:
I believe we've found that `GxB_Matrix_resize` is not thread-safe even on unrelated matrices.